hazelcast/src/test/java/com/hazelcast/internal/dynamicconfig/DynamicConfigTest.java

```diff
@@ -132,6 +132,7 @@ protected HazelcastInstance getDriver() {
     @Test(expected = UnsupportedOperationException.class)
     public void testAddWanReplicationConfigIsNotSupported() {
         WanReplicationConfig wanReplicationConfig = new WanReplicationConfig();
+        wanReplicationConfig.setName(name);
```

Comment:
In this test, to differentiate between `UnsupportedOperationException` thrown from `DynamicConfigurationAwareConfig` or `WanReplicationService#addWanReplicationConfig`, I would suggest removing the `expected` exception definition in `@Test` annotation and assert at a finer level:

```
        UnsupportedOperationException exception =
                Assertions.catchThrowableOfType(() -> getDriver().getConfig().addWanReplicationConfig(wanReplicationConfig),
                UnsupportedOperationException.class);
        assertNotNull(exception);
        assertThat(exception).hasMessage("Adding new WAN config is not supported.");
```

Reply:
Thanks @vbekiaris -- I have made that change in the latest commit.